### PR TITLE
systemvm: FIX haproxy is not reloaded but restarted in VR

### PR DESCRIPTION
**Pull request by ustcweizhou**
## Description

when upgrade to cloudstack 4.14 or 4.15, there is no process ends with "/run/haproxy.pid" after reload haproxy.
so haproxy is reloaded -> restarted -> reloaded -> restarted -> and so on

After restarting haproxy
```
root@r-2040-VM:~# ps aux |grep /usr/sbin/haproxy
root     24894  0.3  1.3  20492 13352 ?        Ss   09:14   0:00 /usr/sbin/haproxy -Ws -f /etc/haproxy/haproxy.cfg -p /run/haproxy.pid
root     24895  0.1  1.1  22332 11372 ?        Ss   09:14   0:00 /usr/sbin/haproxy -Ws -f /etc/haproxy/haproxy.cfg -p /run/haproxy.pid
```

After reloading haproxy
```
root@r-2040-VM:~# systemctl reload haproxy
root@r-2040-VM:~# ps aux |grep /usr/sbin/haproxy
root     24894  0.0  1.3  20492 13300 ?        Ss   09:14   0:00 /usr/sbin/haproxy -Ws -f /etc/haproxy/haproxy.cfg -p /run/haproxy.pid -sf 24895
root     26694  0.0  0.9  22196  9488 ?        Ss   09:16   0:00 /usr/sbin/haproxy -Ws -f /etc/haproxy/haproxy.cfg -p /run/haproxy.pid -sf 24895
```

so use 'grep' to find if haproxy is running.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Review by rhtyd — APPROVED** [review]

File: systemvm/debian/opt/cloud/bin/cs/CsLoadBalancer.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def process(self):
             CsHelper.copy(HAPROXY_CONF_T, HAPROXY_CONF_P)
 
             proc = CsProcess(['/run/haproxy.pid'])
-            if not proc.find():
+            if proc.grep("/usr/sbin/haproxy") == -1:
```
Comment (rhtyd) [inline systemvm/debian/opt/cloud/bin/cs/CsLoadBalancer.py:49]: haven't tested this, should this be !=0 instead @ustcweizhou ?

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: systemvm/debian/opt/cloud/bin/cs/CsLoadBalancer.py (hunk 1 above)
Comment (sureshanaparti) [inline systemvm/debian/opt/cloud/bin/cs/CsLoadBalancer.py:49]: @ustcweizhou this is the same fix done in the PR: https://github.com/apache/cloudstack/pull/4386, changes in "_CsRedundant.py_" for keepalived process. As I've commented in that PR, on a broader view, other processes might have similar issue. can this be fixed in CsProcess.py find() ? else at some point, we can end up replacing all _proc.find()_ with _proc.grep()_.

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: systemvm/debian/opt/cloud/bin/cs/CsLoadBalancer.py (hunk 1 above)
Comment (weizhouapache) [reply systemvm/debian/opt/cloud/bin/cs/CsLoadBalancer.py:49]: @sureshanaparti I will make change on proc.find

**Pull request closed without merging**